File: src/main/java/net/alephdev/calendar/dto/SprintDto.java
package net.alephdev.calendar.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
@Schema(description = "DTO для создания и обновления спринта")
public class SprintDto {
  @Schema(description = "Основная версия спринта", example = "1.0")
  private final String majorVersion;
  
  @Schema(description = "Дата начала спринта", example = "2024-01-01")
  private final LocalDate startDate;
  
  @Schema(description = "Дата окончания спринта", example = "2024-01-15")
  private final LocalDate endDate;
  
  @Schema(description = "Дата начала регрессионного тестирования", example = "2024-01-10")
  private final LocalDate regressionStart;
  
  @Schema(description = "Дата окончания регрессионного тестирования", example = "2024-01-14")
  private final LocalDate regressionEnd;
  
  @Schema(description = "ID команды, ответственной за спринт", example = "1")
  private final Integer teamId;
}

File: angular/src/app/models/dto/sprint-dto.ts
/**
 * DTO для создания или обновления спринта.
 *
 * @property majorVersion - Версия спринта (например, 1.0.0)
 * @property startDate - Дата начала спринта (ISO формат)
 * @property endDate - Дата окончания спринта (ISO формат)
 * @property regressionStart - Дата начала регрессионного тестирования (ISO формат)
 * @property regressionEnd - Дата окончания регрессионного тестирования (ISO формат)
 * @property teamId - Идентификатор команды, которая ведёт спринт
 */
export interface SprintDto {
    majorVersion: string;
    startDate: string;
    endDate: string;
    regressionStart: string;
    regressionEnd: string;
    teamId: number;
}
